pkg/db/lifecycle.go

```diff
@@ -61,19 +62,24 @@ func (m *lifecycleManager) GetId() string {
 	return fmt.Sprintf("db/%s", m.settings.Uri.Database)
 }
 
-func (m *lifecycleManager) Purge(ctx context.Context) error {
-	var err error
+func (m *lifecycleManager) Purge(ctx context.Context) (err error) {
 	var table string
 	var tables []string
 
+	defer func() {
+		if closeErr := m.db.Close(); closeErr != nil {
+			err = errors.Join(err, fmt.Errorf("could not close database: %w", closeErr))
+		}
+	}()
+
 	rows, err := m.db.QueryContext(ctx, "SELECT TABLE_NAME FROM information_schema.TABLES WHERE TABLE_SCHEMA = ?;", m.settings.Uri.Database)
 	if err != nil {
 		return fmt.Errorf("failed to check tables of database: %w", err)
 	}
 
 	for rows.Next() {
 		if err = rows.Scan(&table); err != nil {
-			return fmt.Errorf("could not scan row: %w", err)
+			break
```

Comment:
It is a bit counter-intuitive that we retrieve the error later again by calling rows.Err()... do we need a comment for this?

Reply:
can add a comment..